clang/lib/Driver/Driver.cpp

```diff
@@ -4003,9 +4038,9 @@ bool Driver::checkForSYCLDefaultDevice(Compilation &C,
 
   // Do not do the check if the default device is passed in -fsycl-targets
   // or if -fsycl-targets isn't passed (that implies default device)
-  if (const Arg *A = Args.getLastArg(options::OPT_fsycl_targets_EQ)) {
+  if (const Arg *A = Args.getLastArgNoClaim(options::OPT_fsycl_targets_EQ)) {
```

Comment:
Is this intentionally updated to `getLastArgNoClaim `?

Reply:
Yes.  The reason is to reduce the noise of the diagnostic (i.e. not have the diagnostic be emitted multiple times for `-fsycl-targets` usage).  When the driver is evaluating the arguments to `-fsycl-targets` via `getSYCLDeviceTriple` we claim it then here: https://github.com/intel/llvm/blob/1e45bb44197997cc46acd5a3303f6840e5f5670f/clang/lib/Driver/Driver.cpp#L925